### PR TITLE
59: errorsnack conversion complete

### LINKED ISSUE #59: Update ErrorSnack component
### Tasks
- [x] Folderize
- [x] Create unit tests
- [x] Convert to CSS Modules
- [x] Convert to functional component
- [x] Add PropTypes and Default Props
- [x] Convert to Typescript

### PR DESCRIPTION
**Pull request by Incin**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Header from './components/Header';
 import Body from './components/Body';
 import Footer from './components/Footer';
 import FullscreenButton from './components/FullscreenButton';
-import ErrorSnack from './components/ErrorSnack';
+import ErrorSnack from './components/ErrorSnack/ErrorSnack.tsx';
```
Comment (jefferymills) [inline src/App.js]: Adding the `index.ts` file should make it so this import only has to be `./components/ErrorSnack`

File: src/components/ErrorSnack/ErrorSnack.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { configureStore } from 'redux-mock-store';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
```
Comment (jefferymills) [inline src/components/ErrorSnack/ErrorSnack.test.js:4]: Just FYI, after my current task is merged you won't need to import this in each test file.

File: src/components/ErrorSnack/ErrorSnack.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  test('should render with supplied error message', () => {
+    render(
+      <Provider store={store}>
+        <ErrorSnack errorMsg="" handleClose={() => {}} />
+      </Provider>,
+    );
+    expect(screen.getByText('test error message')).toBeInTheDocument();
+  });
+
+  //   test('should close and remove error message', async () => {
```
Comment (jefferymills) [inline src/components/ErrorSnack/ErrorSnack.test.js:36]: Do we need to keep this around?

File: src/components/ErrorSnack/ErrorSnack.tsx [hunk 4]
```diff
@@ -1,27 +1,30 @@
...
+import { Action, Dispatch } from 'redux';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
 import Snackbar from '@material-ui/core/Snackbar';
 import Button from '@material-ui/core/Button';
+import { setErrorMessage } from '../../actions';
 
-import { setErrorMessage } from '../actions';
+// import styles from './ErrorSnack.module.scss';
```
Comment (jefferymills) [inline src/components/ErrorSnack/ErrorSnack.tsx]: Still need this?

**Review by jefferymills — CHANGES_REQUESTED** [review]
Let's update the path to ErrorSnack in App.js. The other comments are up to your discretion.

File: src/components/ErrorSnack/ErrorSnack.test.js (hunk 3 above)
Comment (Incin) [reply src/components/ErrorSnack/ErrorSnack.test.js:36]: Yeah, that was the one i mentioned on our call where its more of a mental note that it needs to be in the integration test. i will put a comment above it though stating that so it doesnt just look like forgotten code

File: src/components/ErrorSnack/ErrorSnack.tsx (hunk 4 above)
Comment (Incin) [reply src/components/ErrorSnack/ErrorSnack.tsx]: removed

**Review by jefferymills — APPROVED** [review]
🔋

**Pull request merged**